Marlin/configuration_store.cpp

```diff
@@ -64,28 +64,31 @@
  *  195  M206 XYZ  home_offset                      (float x3)
  *  207  M218 XYZ  hotend_offset                    (float x3 per additional hotend)
  *
- * Global Leveling:
+ * Global Leveling:                                 6 bytes
  *  219            z_fade_height                    (float)
+ *  223            GRID_MAX_POINTS_X                (uint8_t)
+ *  224            GRID_MAX_POINTS_Y                (uint8_t)
```

Comment:
Over the longer term we may want to allow variable mesh density as a `G29` option (`P`, `X`, `Y`), so I think each form of leveling should still save its own mesh dimensions — the last ones used. We should then construct EEPROM loading so it works as long as the stored mesh can fit into the configured max dimensions. At that point the mesh dimensions will be made variable.

In the meantime, still keep this separate for each form of leveling, and we'll save the dimensions used, even if they're the fixed configured values.